torch/csrc/lazy/core/tensor_impl.cpp

```diff
@@ -73,14 +77,14 @@ LTCTensorImpl::LTCTensorImpl(LazyTensor&& tensor)
                                           c10::DispatchKey::AutogradLazy},
                       c10::scalarTypeToTypeMeta(tensor.dtype()),
                       backendDeviceToAtenDevice(tensor.GetDevice())),
-      tensor_(std::move(tensor)) {
+      tensor_(c10::make_intrusive<LazyTensor>(std::move(tensor))) {
   // This is a temporary fix for a PyTorch core issue,
   // according to https://github.com/pytorch/xla/pull/2682.
   is_non_overlapping_and_dense_ = false;
 }
 
-void LTCTensorImpl::set_tensor(const LazyTensor& lazy_tensor) {
-  tensor_ = lazy_tensor;
+void LTCTensorImpl::set_tensor(const LazyTensorPtr& lazy_tensor) {
+  tensor_ = c10::make_intrusive<LazyTensor>(*lazy_tensor);
```

Comment:
Case in point: this is super puzzling under the current typing of LazyTensorPtr (you're doing a copy of LazyTensor when you have a perfectly good reference counted pointer). But if it was a UniquePtr (and you took a pointer to LazyTensor directly) the copy would make more sense.